### PR TITLE
feat(frontend): findTwinToken util

### PR DESCRIPTION
**Pull request by DenysKarmazynDFINITY**
# Motivation

The goal is to create a util for finding a twin token that can be reused in token-grouping.util as well as in the upcoming `convert.store`.

### CODE REVIEW

File: src/frontend/src/lib/utils/token.utils.ts [hunk 1]
```diff
@@ -183,3 +185,22 @@ export const sumUsdBalances = ([usdBalance1, usdBalance2]: [
...
+	tokenToPair,
+	tokens
+}: {
+	tokenToPair: Token;
+	tokens: Token[];
+}): IcCkToken | undefined =>
+	isRequiredTokenWithLinkedData(tokenToPair)
+		? (tokens.find(
+				(token) => token.symbol === tokenToPair.twinTokenSymbol && isIcCkToken(token)
+			) as IcCkToken | undefined)
```
Comment (AntonioVentilii) [inline src/frontend/src/lib/utils/token.utils.ts:205]: shouldn't `isIcCkToken` already cast it as `IcCkToken`? is the casting necessary?

`export const isIcCkToken = (token: Token): token is IcCkToken =>...`

File: src/frontend/src/tests/lib/utils/token.utils.spec.ts [hunk 2]
```diff
@@ -275,3 +280,39 @@ describe('sumUsdBalances', () => {
...
+			tokens: [...mockTokens, ckBtcToken],
+			tokenToPair: BTC_MAINNET_TOKEN
+		});
+
+		expect(result).toBe(ckBtcToken);
+	});
+
+	it('should return undefined if no twin token found', () => {
+		const result = findTwinToken({
+			tokens: mockTokens,
```
Comment (AntonioVentilii) [inline src/frontend/src/tests/lib/utils/token.utils.spec.ts:301]: `tokens: [...mockTokens, ckBtcToken]` maybe just to show that there is at least a twin token?

Comment (DenysKarmazynDFINITY) [reply src/frontend/src/tests/lib/utils/token.utils.spec.ts]: done!

File: src/frontend/src/lib/utils/token.utils.ts (hunk 1 above)
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/utils/token.utils.ts:205]: Unfortunately, still necessary.

**Comment by DenysKarmazynDFINITY** [comment]
@AntonioVentilii ready for another look

File: src/frontend/src/lib/utils/token.utils.ts [hunk 3]
```diff
@@ -207,3 +209,22 @@ export const sumUsdBalances = ([usdBalance1, usdBalance2]: [
...
```
Comment (AntonioVentilii) [inline src/frontend/src/lib/utils/token.utils.ts:229]: I just noticed one thing, sorry for the late note:

can you use `IcCkTokenSchema.safeParse` here?

Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/utils/token.utils.ts:229]: I am now using `import { isIcCkToken } from '$icp/validation/ic-token.validation';` to validate if it's a IcCkToken. Is this something you had in mind?

Comment (AntonioVentilii) [reply src/frontend/src/lib/utils/token.utils.ts:229]: even better! thank you!

**Pull request merged**